core/src/mender-client.c

```diff
@@ -485,6 +485,11 @@ mender_commit_artifact_data(void) {
         return MENDER_FAIL;
     }
 
+    if (MENDER_OK != mender_api_artifact_name_load()) {
+        /* Error already logged */
+        return MENDER_FAIL;
+    }
```

Comment:
All,

An alternative of doing this would be to rework both inventory and check for deployments API call to not use at all the in-memory value but always read from the persistent storage.